src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -64,16 +56,80 @@ private void Remove(JobDefinition? jobDefinition)
             return;
         }
 
-        allJobs.Remove(jobDefinition);
-        if (jobDefinition.CronExpression is not null)
+        JobDefinitions.Remove(jobDefinition);
+
+        // TODO: Shouldn't we also remove related entries in DependentJobsPerPrincipalJobType?
```

Comment:
Good point. That might be a bug right now.

Reply:
If that's ok with you, I'd prefer to not deal with this as part of this PR.

I've logged #107 to keep track of it.

Reply:
071Absolutely! Can you reference #107 here instead of just TODO? Might be easier for future us

Reply:
It's already part of the comment. See the line below ;-)